pylint/checkers/classes.py

```diff
@@ -926,9 +926,12 @@ def _check_unused_private_functions(self, node: nodes.ClassDef) -> None:
                     continue
                 if isinstance(attribute.expr, nodes.Name) and attribute.expr.name in (
                     "self",
+                    "cls",
```

Comment:
I think we already fixed something really similar in another MR, we have a duplication between this and ``child.expr.name in ("self", "cls", node.name)`` line 969/972.

Reply:
I can still reproduce #4849 against master. I don't quite understand what the lines you reference fixed.

Reply:
A previous false positive for ``unused-private-member``. I was wondering if we need to refactor that check to have a single piece of code to check that a name represent the current class. It's probably a little late now 😄

Reply:
Ok. Yes, it seems like the possibility to access attributes/methods on `cls` representing the class is beeing added by parts now.
I'm still unsure of hardcoding "cls" into the checks in general without further specification. While pylint checks, that the convention with "cls" as the first arg of a classmethod is followed, it's still no guarantee. And if this rule is disabled on a case by case basis then all these "fixes" will fail again.
But I guess that's the same for the "self" argument/word.

Reply:
https://github.com/PyCQA/pylint/pull/4645 found the MR I was talking about.

Reply:
Ah, I see.
I'm definitely in favor of "a single piece of code to check that a name represent the current class". This would address my above concerns.
But I don't think I'm in the position to judge this  -.-

Reply:
@yushao2 do you have an opinion on this ?

Reply:
@Pierre-Sassoulas kinda rusty, let me refresh my brain for a bit

Reply:
@Pierre-Sassoulas if it can be refactored now, it would be great (since i foresee that it might be refactored again in the future)


Just my two-cents to try to make code more concise and DRY early on rather than doing it later. Could you give it a shot @yumasheta ?

Reply:
I'm humbled, but as this is my first contribution to pylint source code (and first ever look at it), I reckon there are better qualified people out there.
Also the earliest I could do would be Okt/Nov.

Reply:
@yushao2 what do you think about merging this then refactoring in #4965 ? Ideally I'd like to include this in 2.11 that should be released soon.

Reply:
@Pierre-Sassoulas lets do that, i will check on the compatibility once this is merged with my branch